src/index.ts

```diff
@@ -160,36 +164,36 @@ const files: JupyterFrontEndPlugin<void> = {
     tracker: IEditorTracker,
     labShell: ILabShell
   ) => {
-    let _model: any;
+    const handler = new DebuggerHandler<FileHandler>(FileHandler);
+    const activeSessions: {
+      [id: string]: Session.ISession;
+    } = {};
+
     labShell.currentChanged.connect((_, update) => {
       const widget = update.newValue;
-      if (!(widget instanceof FileEditor)) {
+      if (!(widget instanceof DocumentWidget)) {
         return;
       }
 
-      //  Finding if the file is backed by a kernel or attach it to one.
+      const content = widget.content;
+      if (!(content instanceof FileEditor)) {
+        return;
+      }
 
       const sessions = app.serviceManager.sessions;
-
-      void sessions.findByPath(widget.context.path).then(model => {
-        _model = model;
-        const session = sessions.connectTo(model);
-        debug.session.client = session;
-      });
-    });
-
-    app.commands.addCommand(CommandIDs.debugFile, {
-      execute: async _ => {
-        if (!tracker || !tracker.currentWidget) {
-          return;
+      void sessions.findByPath(widget.context.path).then(async model => {
```

Comment:
I think this doesn't handle the case when there is no session for the given path (not backed by a kernel).

Will submit a fix to handle that.

Reply:
Done.